### PR TITLE
ECO-3143: Add functional tests

### PR DESCRIPTION
**Pull request by aleksey-kotsuba**
CR ticket - https://spryker.atlassian.net/browse/ECO-3144

### CODE REVIEW

File: src/SprykerEco/Zed/Heidelpay/Business/Adapter/Payment/DirectDebitPayment.php [hunk 1]
```diff
@@ -57,34 +57,4 @@ public function refund(HeidelpayRequestTransfer $refundRequestTransfer)
 
         return $this->verifyAndParseResponse($directDebitPaymentMethod->getResponse());
     }
-
-    /**
-     * @param \Generated\Shared\Transfer\HeidelpayRequestTransfer $authorizeRequestTransfer
-     *
-     * @return \Generated\Shared\Transfer\HeidelpayResponseTransfer
-     */
-    public function authorize(HeidelpayRequestTransfer $authorizeRequestTransfer): HeidelpayResponseTransfer
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Adapter/Payment/DirectDebitPayment.php]: Why this is removed?

File: src/SprykerEco/Zed/Heidelpay/Persistence/HeidelpayEntityManager.php [hunk 2]
```diff
@@ -27,7 +27,6 @@ public function savePaymentHeidelpayDirectDebitRegistrationEntity(
     ): HeidelpayDirectDebitRegistrationTransfer {
         $paymentHeidelpayDirectDebitRegistrationEntity = $this->getPaymentHeidelpayDirectDebitRegistrationQuery()
             ->filterByRegistrationUniqueId($directDebitRegistrationTransfer->getRegistrationUniqueId())
-            ->filterByFkCustomerAddress($directDebitRegistrationTransfer->getIdCustomerAddress())
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Persistence/HeidelpayEntityManager.php]: same question

File: tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteDebitOnRegistrationTest.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+use SprykerEco\Zed\Heidelpay\Business\HeidelpayBusinessFactory;
+use SprykerEco\Zed\Heidelpay\Business\HeidelpayFacade;
+use SprykerEco\Zed\Heidelpay\Persistence\HeidelpayRepository;
+use SprykerEcoTest\Shared\Heidelpay\HeidelpayTestConfig;
+use SprykerEcoTest\Zed\Heidelpay\Business\DataProviders\PaymentBuilder;
+use SprykerEcoTest\Zed\Heidelpay\Business\Mock\SuccessfulResponseHeidelpayBusinessFactory;
+
+/**
+ * @group Functional
+ * @group Spryker
```
Comment (gechetspr) [inline tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteDebitOnRegistrationTest.php]: it looks like it should be SprykerEcoTest

File: tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteDebitOnRegistrationTest.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    protected function createSuccessfulPaymentHeidelpayFactoryMock(): HeidelpayBusinessFactory
+    {
+        return new SuccessfulResponseHeidelpayBusinessFactory();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Heidelpay\Persistence\HeidelpayRepository
+     */
+    protected function createRepository(): HeidelpayRepository
```
Comment (gechetspr) [inline tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteDebitOnRegistrationTest.php]: you can return an interface here

File: tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteDebitOnRegistrationTest.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+{
+    /**
+     * @return void
+     */
+    public function testSuccessfulExecuteRefund(): void
+    {
+        //Arrange
+        $salesOrder = $this->createSuccessOrder();
+        $factory = $this->createSuccessfulPaymentHeidelpayFactoryMock();
+        $heidelpayFacade = (new HeidelpayFacade())
```
Comment (gechetspr) [inline tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteDebitOnRegistrationTest.php]: You might want to put this in the separate method.

File: tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteRefundTest.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+use SprykerEco\Zed\Heidelpay\Business\HeidelpayBusinessFactory;
+use SprykerEco\Zed\Heidelpay\Business\HeidelpayFacade;
+use SprykerEco\Zed\Heidelpay\Persistence\HeidelpayRepository;
+use SprykerEcoTest\Shared\Heidelpay\HeidelpayTestConfig;
+use SprykerEcoTest\Zed\Heidelpay\Business\DataProviders\PaymentBuilder;
+use SprykerEcoTest\Zed\Heidelpay\Business\Mock\SuccessfulResponseHeidelpayBusinessFactory;
+
+/**
+ * @group Functional
+ * @group Spryker
```
Comment (gechetspr) [inline tests/SprykerEcoTest/Zed/Heidelpay/Business/HeidelpayFacadeExecuteRefundTest.php]: please check all test files

File: src/SprykerEco/Zed/Heidelpay/Business/Adapter/Payment/DirectDebitPayment.php (hunk 1 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Zed/Heidelpay/Business/Adapter/Payment/DirectDebitPayment.php]: Authorization is not part of our scope. We are going to have implementation based immediate debit after registration.

File: src/SprykerEco/Zed/Heidelpay/Persistence/HeidelpayEntityManager.php (hunk 2 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Zed/Heidelpay/Persistence/HeidelpayEntityManager.php]: A bug was discovered during tests implementation. In previous version I had duplication of registration rows after updating row with customer address.

**Pull request merged**